File: pset6/psets/sentimental-readability/readability.py
from cs50 import get_string


def main():
    input = get_string("Text: ")

    L = (count_letters(input)/count_words(input)) * 100
    S = (count_sentences(input)/count_words(input)) * 100

    # calculate the index
    index = round(0.0588 * L - 0.296 * S - 15.8)

    # calculate the grade level
    if (index >= 16):
        print("Grade 16+")
    elif (index < 1):
        print("Before Grade 1")
    else:
        print(f"Grade {index}")

# count the total num of letters in the input


def count_letters(input):
    count = 0
    for i in range(len(input)):
        if (input[i].isalpha()):
            count += 1
    return count


# count the total num of words starting with 1
def count_words(input):
    space = " "
    count = 1
    for i in range(len(input)):
        if (input[i] == space):
            count += 1
    return count


# count the total sentences in input
def count_sentences(input):
    count = 0
    for i in range(len(input)):
        if (input[i] == '.' or input[i] == '!' or input[i] == '?'):
            count += 1
    return count


# start the program with the current module
if __name__ == "__main__":
    main()

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

// setting up the prototype for the function we are using
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string prompt = get_string("Text: ");

    // L and S represent average number of letters per 100 words and sentences resp.
    float L = ((float)  count_letters(prompt) / (float) count_words(prompt)) * 100;
    float S = ((float) count_sentences(prompt) / (float) count_words(prompt)) * 100;
    // formula to calculate the grade level
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int average = round(index);

    // only giving the grade level if it's between 1-16.
    if (average >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (average < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n",  average);
    }
}

// function to calculate the number of letters in the given text
int count_letters(string text)
{
    int count = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]))
        {
            count++;
        }
    }
    return count;
}

// function to calculate the number of words
int count_words(string text)
{
    // Initializing words as 1 because the first word is counted as one
    int count = 1;
    string space = " ";
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == ' ')
        {
            count++;
        }
    }
    return count;
}

// function to calculate the number of sentences in the text
int count_sentences(string text)
{
    int count = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count++;
        }
    }
    return count;
}